### DISCUSSION TITLE
PHP ERROR using /api/v2/services/dhcp_server/static_mapping

### DISCUSSION THREAD

**Discussion by couraillon** [jaredhendrickson13/pfsense-api#616] [Q&A] [answered]
Hello,

I'm using pfSense v24.03 and the pfsense-api v2.2.2 to remove and add some static_mappings.
My script should read two lists of MAC addresses, those to be removed from static_mappings and those to be added.
The script first checks whether each MAC address is already registered, and that part works well. But then when it comes to delete all the MAC addresses I get the following error:

`PHP ERROR: Type: 1, File: /usr/local/pkg/RESTAPI/Core/Model.inc, Line: 2289, Message: Allowed memory size of 536870912 bytes exhausted (tried to allocate 8192 bytes) @ 2024-12-13 11:49:24`

I have read this [discussion](https://github.com/jaredhendrickson13/pfsense-api/discussions/520), but the error still seems to exist on my side, despite the patch.
Is this a known error when there are too many queries in a row?

Thank you!

**Comment by jaredhendrickson13** [comment]
I have not seen this particular issue before. Roughly how many DHCP servers and static mappings exist in your config currently and how many requests per second are you making to the API?

**Reply by couraillon** [reply]
I have 13 DHCP servers and about 450 static mappings on the interface on which I'm trying to add/remove static mappings.
The error occurs on the first attempt to delete one static mapping. I'm not sure about the execution speed of my script but I can add delays between each request if you think that's where it might come from.
Thank you!

**Reply by jaredhendrickson13** [reply]
It's more likely there is an issue with the API loading some objects recursively on this endpoint.  I will do some testing and see if I can reproduce the issue and find a definite cause.

**Reply by jaredhendrickson13** [reply, marked as answer]
I was able to reproduce the issue and pushed a fix that will be included in v2.3.0. I created issue #617 for the bug itself. Thanks for reporting this!

**Reply by couraillon** [reply]
You're welcome, thank YOU for your work!! :)
